flybywiresim/a32nx
# feat(efb): automatically load interior lights preset

Fixes #6723 

## Summary of Changes

Add the ability to automatically load an interior lighting preset when the aircraft powers up. Which preset is loaded can be configured on the 'Interior Lighting Presets' page of the EFB; there is 3 configurable settings based on the sim's time of day. The feature can be turned off from the same page. It is off by default.

## Screenshots (if necessary)


Old interface:
![pre_efb-auto-load](https://user-images.githubusercontent.com/84216707/185783637-0e2c5070-0658-4ff1-9619-3d0c4ddd32b8.JPG)

New interface:
![post_efb-auto-load](https://user-images.githubusercontent.com/84216707/185783642-25f3e263-f08d-4c3b-a65a-6a1add0c31ce.JPG)

## References





## Additional context

First attempt at contributing to the project, so kind of just wanted to dip my toes in and play around. Happy to accept feedback or change how the implementation has been done / how it looks.


Discord username (if different from GitHub): SpiritZephyr#2349

## Testing instructions

Core Functionality:

1. Load aircraft in a cold and dark state.
2. Go to the presets -> interior lighting presets page of the EFB
3. Setup some lighting presets if you don't have any
4. Configure the automatic loading of presets for day, dawn/dusk and night
5. Enable the automatic loading of presets
6. Set the time of the simulator to an hour which is day time
7. Power on the aircraft (AC power needed)
8. Verify that the correct lighting preset automatically applies to the aircraft
9. Power off the aircraft
10. Repeat steps 7-9 for dawn/dusk and night time
11. Change the selected preset for the current time of day and toggle the 'Enable automatic loading' setting off and on
12. Verify that the new lighting preset was loaded
14. Change the selected preset back to 'None', cycle power to the aircraft or toggle the 'Enable automatic loading' setting
15. Verify that no change was made to the lights.

Check for unintended side-effects after change to SelectInput component (i.e dropdown menus):

1. Confirm that other dropdown menu's (e.g. 'Settings' -> 'Flypad' page, 'TOD Calculator' page) still have the expected default value.


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
